kafka/src/main/java/org/apache/calcite/adapter/kafka/KafkaTableFactory.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.kafka;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.schema.TableFactory;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Implementation of {@link TableFactory} for Kafka. Currently a Kafka topic is mapping
+ * to a STREAM table.
+ */
+public class KafkaTableFactory implements TableFactory<KafkaStreamTable> {
+  public KafkaTableFactory() {
+  }
+
+  /** Creates a Table.
+   *  @param schema Schema this table belongs to
+   * @param name Name of this table
+   * @param operand The "operand" JSON property
+   * @param rowType Row type. Specified if the "columns" JSON property.
+   */
+  @Override public KafkaStreamTable create(SchemaPlus schema,
+      String name,
+      Map<String, Object> operand,
+      RelDataType rowType) {
+    KafkaTableOptions tableOptionBuilder = new KafkaTableOptions();
+
+    tableOptionBuilder.setBootstrapServers(
+        (String) operand.getOrDefault(KafkaTableConstants.SCHEMA_BOOTSTRAP_SERVERS, null));
+    tableOptionBuilder.setTopicName(
+        (String) operand.getOrDefault(KafkaTableConstants.SCHEMA_TOPIC_NAME, null));
+
+    KafkaRowConverter rowConverter = null;
+    if (operand.containsKey(KafkaTableConstants.SCHEMA_ROW_CONVERTER)) {
+      try {
+        rowConverter = (KafkaRowConverter) Class.forName(
+            (String) operand.get(KafkaTableConstants.SCHEMA_ROW_CONVERTER))
+            .newInstance();
+      } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      rowConverter = new KafkaRowConverterImpl();
+    }
+    tableOptionBuilder.setRowConverter(rowConverter);
+
+    if (operand.containsKey(KafkaTableConstants.SCHEMA_CONSUMER_PARA)) {
+      tableOptionBuilder.setConsumerParams((Map) operand.get(
+          KafkaTableConstants.SCHEMA_CONSUMER_PARA));
+    }
+    if (operand.containsKey(KafkaTableConstants.SCHEMA_CUST_CONSUMER)) {
+      try {
+        tableOptionBuilder.setConsumer(
+            (Consumer) Class.forName((String) operand.get(KafkaTableConstants.SCHEMA_CUST_CONSUMER))
+                .getConstructor(OffsetResetStrategy.class)
+                .newInstance(OffsetResetStrategy.NONE)
+        );
+      } catch (ClassNotFoundException e) {
+        throw new RuntimeException(
+            String.format(Locale.getDefault(), "Class '%s' is not found.",
```

Comment:
What I meant here is more like
```java
catch (CheckedExeption e) {
  final String details = String.format("Failed to create table %s (topic %s) in schema %s", name, topicName, schema,getName());
   throw new RuntimeException(details, e); // instead of just: throw new RuntimeException(e);
}
```

You don't need to catch each exception individually.

Reply:
ah I got you, let me update it